### ISSUE TITLE
supported=1 detected=0

### ISSUE THREAD

**Issue by tambetm** [ivmech/ivport-v2#10]
I have IVPORT DUAL V2 connected to two V2.1 cameras on RPi 3 model B. I can detect both ivport dual and camera module successfully with i2cdetect:
```
tamme@tamme-desktop:~/ivport-v2$ i2cdetect -y 1
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: 10 -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- 64 -- -- -- -- -- -- -- -- -- -- -- 
70: 70 -- -- -- -- -- -- --                         
```
But when I test for cameras they are not detected:
```
tamme@tamme-desktop:~/ivport-v2$ vcgencmd get_camera
supported=1 detected=0
```
What could be causing this? Internet suggests to check the ribbon cables, but I've triple-checked all cables dozen times and pretty confident they should be ok. Also both cameras work when connected directly to RPi.

Could it be that I didn't bother to solder the ivport card to gpio connectors? I just wanted to test it first and tilted it in such a way that all pins should be connected. The setup looks like this:

![img_2421](https://user-images.githubusercontent.com/3816322/32970964-d4cd2a08-cbf3-11e7-9f55-2325b4d646ce.JPG)

**Comment by tambetm** [comment]
I soldered the board properly, but it still gave the same error. Then I did a reboot and it worked! So far it seems that:
1. I need to boot up raspberry.
2. Run init_ivport.py.
3. Reboot the system.
4. No need to run init_ivport.py second time, `vcgencmd get_camera` just works.

Not sure if the soldering helped or I just forgot to reboot the last time.

**Comment by JamesCordell** [comment]
I found this out the hard way. Have you managed to git it working with out the reboot? this is a bit of a pain and not mentioned in the manual!

**Comment by nortont** [comment]
This worked for me - do a reboot after init_ivport
I was restarting thinking that was the same

**Comment by leoshmu** [comment]
@tambetm I have the exact same issue - I'm using a v2 dual with v2.1 cameras. I have connected it correctly as far as I can tell and my cameras alone work perfectly when connected to the rasbperry pi 4. The addresses appear correctly but the detected = 0. Perhaps it's a problem due to incompatibility with the raspberry pi 4? I'm really hoping someone can shed light on why it's not working!

**Comment by tambetm** [comment]
@leoshmu unfortunately I don't have any new information. AFAIK the kernel needed the device to be activated, it couldn't detect it before. So I needed to run `init_ivport.py` first and then restart, then the device was visible to the kernel and was detected.

**Comment by mar97co** [comment]
Hi everyone, it’s been a long time, but for a university course project, I dug out an IVPORT-V2 QUAD that I’d never used and had been keeping in a drawer. 
I tried everything and **_I found a solution_** that works with a **Raspberry Pi** 4 running [**2023-05-03-raspios-bullseye-armhf.img.xz**](https://downloads.raspberrypi.com/raspios_armhf/images/raspios_armhf-2023-05-03/2023-05-03-raspios-bullseye-armhf.img.xz), **IVPORT-V2 QUAD** and **Raspberry Pi Camera V2.1**.

I followed the steps as described in the README, but I kept getting this error:
```
python3 test_ivport_quad.py
mmal: Cannot read camera info, keeping the defaults for OV5647
mmal: mmal_vc_component_create: failed to create component 'vc.ril.camera' (1:ENOMEM)
mmal: mmal_component_create_core: could not create component 'vc.ril.camera' (1)
mmal: Failed to create camera component
mmal: main: Failed to create camera component
mmal: Camera is not detected. Please check carefully the camera module is installed correctly

mmal: Cannot read camera info, keeping the defaults for OV5647
mmal: mmal_vc_component_create: failed to create component 'vc.ril.camera' (1:ENOMEM)
mmal: mmal_component_create_core: could not create component 'vc.ril.camera' (1)
mmal: Failed to create camera component
mmal: main: Failed to create camera component
mmal: Camera is not detected. Please check carefully the camera module is installed correctly
```
Basically, when the Raspberry Pi boots up, it tries to initialize the connected camera, but because the multiplexer is in the way, it can’t find it.
**Ho semplicemente fatto questo:**
1. Disconnected the power from the Raspberry Pi.
2. Connected the camera directly to the Raspberry Pi.
3. Plugged it back in and waited for the boot process to complete.
4. While it was running, I abruptly disconnected the camera and connected it to the IVPORT. Then I connected the IVPORT to the Raspberry Pi.
5. I ran the commands `python3 init_ivport.py` and then `python3 test_ivport_quad.py`
6. Et voilà! The images captured by both cameras had appeared in the folder!

Unfortunately, you'll have to repeat the process after rebooting...

I hope this is still helpful to someone! Bye! :)
